### PR TITLE
Spatial relationships sample

### PR DESCRIPTION
**Pull request by nCastle1**
This sample demonstrates how to use geometry engine to evaluate the relationships between geometries.

### CODE REVIEW

**Comment by nCastle1** [comment]
@dkemlage Can you please review this? Thanks!

**Comment by dkemlage** [comment]
@nCastle1 Can you update your Pull Request (PR) to only contain the file changes for the new 'Spatial Releationships' sample? I want to add some in-line code comments for the review but the web interface for the Review is not able to locate the files I want to make comments on because the number of files changed is too large.

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/GeometryEngine/SpatialRelationships/SpatialRelationships.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+namespace ArcGISRuntime.WPF.Samples.SpatialRelationships
+{
+    [ArcGISRuntime.Samples.Shared.Attributes.Sample(
+        "Spatial relationships",
+        "GeometryEngine",
+        "This sample demonstrates how to use the GeometryEngine to evaluate the spatial relationships (for example, polygon a contains line b) of between geometries.",
+        "Tap a graphic to select it. The display will update to show the relationships with the other graphics.")]
+    public partial class SpatialRelationships
+    {
```
Comment (dkemlage) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/GeometryEngine/SpatialRelationships/SpatialRelationships.xaml.cs]: Add code comments for the member variables in the WPF, UWP, and Xamarin.Forms projects. In Android and iOS, the following code comment is available:
```
        // References to the graphics and graphics overlay
```

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/GeometryEngine/SpatialRelationships/SpatialRelationships.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+using Esri.ArcGISRuntime.UI.Controls;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace ArcGISRuntime.WPF.Samples.SpatialRelationships
+{
+    [ArcGISRuntime.Samples.Shared.Attributes.Sample(
+        "Spatial relationships",
+        "GeometryEngine",
+        "This sample demonstrates how to use the GeometryEngine to evaluate the spatial relationships (for example, polygon a contains line b) of between geometries.",
```
Comment (dkemlage) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/GeometryEngine/SpatialRelationships/SpatialRelationships.xaml.cs]: Remove the 'of' in the sample description of the Readme.md and Attributes of each SDK sample:
```
"This sample demonstrates how to use the GeometryEngine to evaluate the spatial relationships (for example, polygon a contains line b) between geometries."
```

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/GeometryEngine/SpatialRelationships/SpatialRelationships.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+            };
+
+            // Create the polygon
+            Polygon polygonGeometry = new Polygon(polygonPoints);
+
+            // Define and apply the symbology
+            SimpleLineSymbol polygonOutlineSymbol = new SimpleLineSymbol(SimpleLineSymbolStyle.Solid, System.Windows.Media.Colors.Green, 2);
+            SimpleFillSymbol polygonFillSymbol = new SimpleFillSymbol(SimpleFillSymbolStyle.ForwardDiagonal, System.Windows.Media.Colors.Green, polygonOutlineSymbol);
+
+            // Create the graphic and add it to the graphics overlay
```
Comment (dkemlage) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/GeometryEngine/SpatialRelationships/SpatialRelationships.xaml.cs]: Clarify some code comments (done for WPF - applies for the other SDK's as well):
```
// Create the polygon graphic and add it to the graphics overlay
```

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/GeometryEngine/SpatialRelationships/SpatialRelationships.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+                new MapPoint(-6928550.398185, 2087936.739807),
+                new MapPoint(-3149463.800709, 1840803.011362),
+                new MapPoint(-1563689.043184, 3714900.452072),
+                new MapPoint(-3180355.516764, 5619889.608838)
+            };
+
+            // Create the polygon
+            Polygon polygonGeometry = new Polygon(polygonPoints);
+
+            // Define and apply the symbology
```
Comment (dkemlage) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/GeometryEngine/SpatialRelationships/SpatialRelationships.xaml.cs]: Clarify some code comments (done for WPF - applies for the other SDK's as well):
```
// Define the symbology of the polygon
```

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/GeometryEngine/SpatialRelationships/SpatialRelationships.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+                new MapPoint(-4354240.726880, -609939.795721),
+                new MapPoint(-3427489.245210, 2139422.933233),
+                new MapPoint(-2109442.693501, 4301843.057130),
+                new MapPoint(-1810822.771630, 7205664.366363)
+            };
+
+            // Create the polyline
+            Polyline polylineGeometry = new Polyline(polylinePoints);
+
+            // Create the polyline and add it to the overlay
```
Comment (dkemlage) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/GeometryEngine/SpatialRelationships/SpatialRelationships.xaml.cs]: Clarify some code comments (done for WPF - applies for the other SDK's as well):
```
// Create the polyline graphic and add it to the graphics overlay
```

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/GeometryEngine/SpatialRelationships/SpatialRelationships.xaml.cs [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+            _polylineGraphic = new Graphic(polylineGeometry, new SimpleLineSymbol(SimpleLineSymbolStyle.Dash, System.Windows.Media.Colors.Red, 4));
+            _graphicsOverlay.Graphics.Add(_polylineGraphic);
+
+            // Create the point that defines the point graphic
```
Comment (dkemlage) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/GeometryEngine/SpatialRelationships/SpatialRelationships.xaml.cs]: Clarify some code comments (done for WPF - applies for the other SDK's as well):
```
// Create the map point that defines the point graphic
```

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/GeometryEngine/SpatialRelationships/SpatialRelationships.xaml.cs [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+            MapPoint point = new MapPoint(-4487263.495911, 3699176.480377, SpatialReferences.WebMercator);
+
+            // Define the symbology for the point
+            SimpleMarkerSymbol locationMarker = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle.Circle, System.Windows.Media.Colors.Blue, 10);
+
+            // Create the graphic and add it to the overlay
```
Comment (dkemlage) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/GeometryEngine/SpatialRelationships/SpatialRelationships.xaml.cs]: Clarify some code comments (done for WPF - applies for the other SDK's as well):
```
// Create the point graphic and add it to the graphics overlay
```

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/GeometryEngine/SpatialRelationships/SpatialRelationships.xaml.cs [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+
+            // Create the polyline
+            Polyline polylineGeometry = new Polyline(polylinePoints);
+
+            // Create the polyline and add it to the overlay
+            _polylineGraphic = new Graphic(polylineGeometry, new SimpleLineSymbol(SimpleLineSymbolStyle.Dash, System.Windows.Media.Colors.Red, 4));
+            _graphicsOverlay.Graphics.Add(_polylineGraphic);
+
+            // Create the point that defines the point graphic
+            MapPoint point = new MapPoint(-4487263.495911, 3699176.480377, SpatialReferences.WebMercator);
```
Comment (dkemlage) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/GeometryEngine/SpatialRelationships/SpatialRelationships.xaml.cs]: Change the name of the variable 'point' to be 'pointGeometry'. This matches the names of the 
other variables you created called 'polygonGeometry' and 'polylineGeometry' (done for WPF - applies for the other SDK's as well)

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/GeometryEngine/SpatialRelationships/SpatialRelationships.xaml.cs [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+            }
+
+            // Expand the tree view
+            PolylineTreeEntry.IsExpanded = true;
+            PolygonTreeEntry.IsExpanded = true;
+            PointTreeEntry.IsExpanded = true;
+        }
+
+        /// <summary>
+        /// Returns a list of spatial relationships between two points
```
Comment (dkemlage) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/GeometryEngine/SpatialRelationships/SpatialRelationships.xaml.cs]: Clarify some code comments (done for WPF - applies for the other SDK's as well):
```
/// Returns a list of spatial relationships between two geometries
```

**Comment by nCastle1** [comment]
Thanks for the review!

I kept the tree view for WPF to be consistent with other SDKs. The text trick was needed to work around the lack of a tree view on mobile.

**Pull request merged**